tests/template_tests/test_loaders.py

```diff
@@ -238,6 +249,24 @@ def test_case_sensitivity(self):
             check_sources('index.html', ['/dir1/index.html', '/DIR2/index.html'])
             check_sources('/DIR1/index.HTML', ['/DIR1/index.HTML'])
 
+    def test_file_does_not_exist(self):
+        with self.assertRaises(TemplateDoesNotExist):
+            self.engine.get_template('doesnotexist.html')
+
+    @unittest.skipIf(
+        sys.platform == 'win32',
+        "Python on Windows doesn't have working os.chmod().",
+    )
+    def test_permissions_error(self):
+        path = os.path.join(TEMPLATE_DIR, 'permission.html')
+        with self.set_permissions(path, "222"):
```

Comment:
Please don't write to files in the current checkout. That's bad hygiene in general and also making something I'm working on more difficult :-) Can you keep the logic that used a temporary file?